util/testutil/shell.go

```diff
@@ -88,41 +88,79 @@ func (r *TestingReporter) Stderr() io.Writer {
 	return TestingL.Writer()
 }
 
+// LogMonitor manages a list of functions that should scan lines coming from stdout and stderr Readers
+type LogMonitor struct {
+	monitorFuncs map[string]func(string)
+}
+
+// NewLogMonitor creates a LogMonitor for a given pair of stdout and stderr Readers
+func NewLogMonitor(r shell.Reporter, stdout, stderr io.Reader) *LogMonitor {
+	m := &LogMonitor{}
+	m.monitorFuncs = make(map[string]func(string))
+	go m.ScanLog(io.TeeReader(stdout, r.Stdout()))
+	go m.ScanLog(io.TeeReader(stderr, r.Stderr()))
```

Comment:
When/how these goroutines will exit?

Reply:
`for scanner.Scan() {` will terminate when the scanner reaches EOF. I believe EOF will occur on stdout and stderr when the spawned process terminates.